components/server/src/workspace/config-provider.ts

```diff
@@ -169,36 +171,33 @@ export class ConfigProvider {
                     customConfigString = content;
                     // We do not only care about the config itself but also where we got it from
                     configBasePath = basePath;
-                    fromDefinitelyGp = true;
+                    origin = "definitely-gp";
+                }
+
+                if (!customConfigString) {
+                    // if there's still no configuration, let's infer one
+                    customConfigString = await inferredConfig;
+                    origin = "derived";
```

Comment:
Super-nit: Technically, I think this should be conditional on the inferrer returning something (however, in the current state of the code, this doesn't matter because of the `if (!customConfig) { return undefined; }` below).

```suggestion
                    if (customConfigString) {
                        origin = "derived";
                    }
```